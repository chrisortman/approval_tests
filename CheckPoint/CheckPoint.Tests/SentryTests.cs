using System;
using NUnit.Framework;
using ApprovalTests;
using TypeMock;
using System.Collections.Generic;
using System.Reflection;
using TypeMock.ArrangeActAssert;

namespace CheckPoint.Tests
{
	[TestFixture]
	public class SentryTests
	{
		[Test]
		[Isolated]
		public void GetListOfCalls()
		{
			Sentry s = Monitor.Calls<Model>();

			var controller = new Controller();
			controller.Action();

			Approvals.Approve(s.Results);
		}

		[Test]
		[Isolated]
		public void GetInteractions()
		{
			Sentry s = Monitor.Interactions(typeof(Controller));

			var controller = new Controller();
			controller.Action();

			Approvals.Approve(s.Results);
		}
	}
}