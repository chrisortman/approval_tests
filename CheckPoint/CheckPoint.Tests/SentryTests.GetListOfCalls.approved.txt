CheckPoint.Tests.Model..ctor
CheckPoint.Tests.Model.Call1
CheckPoint.Tests.Model.Call2
